java-generator/it/src/it/preserve-unknown-objects/src/test/java/io/fabric8/it/certmanager/TestPreserveUnkonwnUsability.java

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.fabric8.it.certmanager;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.IntNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.dapr.v1alpha1.Component;
+import io.dapr.v1alpha1.ComponentSpec;
+import io.dapr.v1alpha1.componentspec.Metadata;
+import io.fabric8.kubernetes.api.model.ObjectMeta;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.junit.jupiter.api.Test;
+import io.fabric8.java.generator.testing.KubernetesResourceDiff;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class TestPreserveUnknownUsability {
+
+  @Test
+  void testDeserialization() {
+    // Arrange
+    Component sample =
+      Serialization.unmarshal(getClass().getResourceAsStream("/sample.yaml"), Component.class);
+
+    // Act
+    List<Metadata> metadataList = sample.getSpec().getMetadata();
+
+    // Assert
+    assertEquals(5, metadataList.size());
+  }
+
+  private Component createSampleComponent() throws Exception {
+    Component component = new Component();
+    ComponentSpec spec = new ComponentSpec();
+    spec.setType("pubsub.mqtt");
+    spec.setVersion("v1");
+
+    Metadata meta1 = new Metadata();
+    meta1.setName("consumerID");
+    meta1.setValue(new TextNode("{uuid}"));
+
+    Metadata meta2 = new Metadata();
+    meta2.setName("url");
+    meta2.setValue(new TextNode("tcp://admin:public@localhost:1883"));
```

Comment:
Just to add to my comment regarding the chosen implementation strategy, here - and in the other `set(...)` statements below - is where the mixing between the JSON definition and the Java model caught my eye.

Reply:
You are 100% right, having to deal with `JsonNode` things is ugly and error-prone, but I do not have a better encoding that guarantees the necessary expressiveness and improves the Java integration.
Any idea, alternative, or suggestion would be highly appreciated!

Reply:
At first, when I looked to the original poster issue, I immediately though about `Map`. But I must admit I haven't looked enough into the implications for this kind of solution.
ATM, is my opinion as well that the `JsonNode` based solution is the best tradeoff.

Reply:
👍